pkg/client/rp/verifier.go

```diff
@@ -21,69 +21,71 @@ type IDTokenVerifier interface {
 
 // VerifyTokens implement the Token Response Validation as defined in OIDC specification
 // https://openid.net/specs/openid-connect-core-1_0.html#TokenResponseValidation
-func VerifyTokens(ctx context.Context, accessToken, idTokenString string, v IDTokenVerifier) (oidc.IDTokenClaims, error) {
-	idToken, err := VerifyIDToken(ctx, idTokenString, v)
+func VerifyTokens[C oidc.IDClaims](ctx context.Context, accessToken, idToken string, v IDTokenVerifier) (claims C, err error) {
+	var nilClaims C
```

Comment:
correct. Because of type parameters, I can't pass a direct `nil`. For example an implementation of `oidc.IDClaims` might not be a pointer receiver but a struct and a `nil` would be illigal. Hence, Go dis-allows the usage of `nil` for a generic type.